internal/payloads/error_payload.go

```diff
@@ -49,10 +41,10 @@ func NewInvalidRequestError(ctx context.Context, err error) *ResponseError {
 	}
 	return &ResponseError{
 		HTTPStatusCode: 400,
+		Name:           "invalid request error",
 		Message:        msg,
-		RequestId:      ctxval.TraceId(ctx),
-		Err:            err,
-		Context:        ctx,
+		TraceId:        ctxval.TraceId(ctx),
+		Error:          err,
 	}
```

Comment:
Can you include the actual "Name" in error message too for all relevant errors?

```go
func NewInvalidRequestError(ctx context.Context, err error) *ResponseError {
    name := "invalid request error"
	err = fmt.Errorf("%s: %w", name, err)
	if logger := ctxval.Logger(ctx); logger != nil {
		logger.Warn().Err(err).Msg(msg)
	}
	return &ResponseError{
		HTTPStatusCode: 400,
		Name:           name,
		Message:        name, // you can do this for most errors when name is appropriate
		Error:            err,
		TraceId:        ctxval.TraceId(ctx),
	}
```

Reply:
I thought I might add a 'message' parameter later and pass it from the calling function with a more specific message than just an "invalid request error". Possibly something like "invalid request: parameter missing" or something like that. That might be more useful for the user. Adding parameters would force changing more files, so I planned to make these changes in an upcoming PR.
Does this make sense?